projects/smart-table/src/lib/smart-table/smart-table.component.ts

```diff
@@ -293,4 +315,44 @@ export class SmartTableComponent implements AfterViewInit {
     public toggleOptionalFilters() {
         this.optionalFiltersVisible = !this.optionalFiltersVisible;
     }
+
+    public exportToExcel() {
+        this.pageChanging = true;
+        this.dataService.getData(this.apiUrl, this.httpHeaders, this.dataQuery)
+            .subscribe(data => {
+                const exportData = this.filterOutColumns(data._embedded.resourceList);
+                this.dataService.exportAsExcelFile(exportData, 'smart-table');
+                this.pageChanging = false;
+            });
+    }
+
+    public onColumnsSelected() {
+        const clonedConfiguration = Object.assign({}, this.configuration);
+        clonedConfiguration.columns = clonedConfiguration.columns.map(col => {
+            col.visible = !this.selectableColumns.find(sCol => sCol.value === col.key).hidden;
+            return col;
+        });
+        if (this.configuration.options.persistTableConfig) {
+            if (!this.getLocalStorageKey()) {
+                throw new Error("No 'storageIdentifier' was set to be able to persist table configuration. Please set an unique `storageIdentifier` in your BFF configuration.");
+            }
+            this.localstorageService.setItem(this.getLocalStorageKey(), clonedConfiguration);
```

Comment:
ik zou niet de hele table config persisten. Je komt dan in de problemen als die server-side wijzigt (nieuwe set kolommen na een upgrade). Ik zou enkel de column visibility state bewaren en dan bij het inladen van de server-side config die mergen daarmee. Dan heb je ook die nogal lelijke LOCAL_STORAGE_KEY niet meer nodig, omdat je gewoon getLocalStorageKey() kan oproepen telkens je de key nodig hebt.

Reply:
Read me is uitgebreid en LOCAL_STORAGE_KEY constant is weg.